packages/react/src/components/UIShell/HeaderContainer.js

```diff
@@ -14,6 +14,14 @@ const HeaderContainer = ({ isSideNavExpanded, render: Children }) => {
   const [isSideNavExpandedState, setIsSideNavExpandedState] =
     useState(isSideNavExpanded);
 
+  // Closing the SideNav when the 'ESC' key is pressed
+  document.addEventListener('keydown', function (event) {
```

Comment:
I think this eventListener needs to be cleaned up on unmount, the `useEvent` hook (packages/react/src/internal/useEvent.js) should be able to handle all that for you 
<img width="548" alt="image" src="https://user-images.githubusercontent.com/40550942/232601364-adca736e-eb51-4d9a-8f33-d894b5126d43.png">

Also, is there an element we can attach this event to or does it need to be the whole document? Like, could we add `onKeyDown` events to the `Children` instead?

Reply:
I'm gonna check if we can attach to an element to improve that

Reply:
I changed to `useEvent` hook on `HeaderContainer` component and attach to an eventHandler on `SideNav` component. 
Thanks for the suggestions, now looks better!